src/e2eTest/utils/actions/custom-actions/createCase.action.ts

```diff
@@ -1,30 +1,82 @@
+import { Page } from '@playwright/test';
+import { initIdamAuthToken, initServiceAuthToken, getUser } from 'utils/helpers/idam-helpers/idam.helper';
+import { performAction, performActions } from '@utils/controller';
+import {createCase} from "@data/page-data/createCase.page.data";
+import {addressDetails} from "@data/page-data/addressDetails.page.data";
 
-
-let caseInfo: { id: string; fid: string, state: string };
+let caseInfo: { id: string; fid: string; state: string };
 const testConfig = TestConfig.ccdCase;
 
 export class CreateCaseAction implements IAction {
   private eventToken?: string;
+  constructor(private readonly axios: AxiosInstance = Axios.create()) {}
 
-  constructor(
-    private readonly axios: AxiosInstance = Axios.create()
-  ) {
+  async execute(page: Page, action: string, fieldName: actionData, data?: actionData): Promise<void> {
+    const actionsMap = new Map<string, () => Promise<void>>([
+      ['createCase', () => this.createCaseAction(page, action, fieldName, data)],
+      ['selectAddress', () => this.selectAddress(fieldName)],
+      ['selectLegislativeCountry', () => this.selectLegislativeCountry(fieldName)],
+      ['selectClaimantType', () => this.selectClaimantType(fieldName)],
+      ['selectCaseOptions', () => this.selectCaseOptions()],
+      ['enterAddress', () => this.enterTestAddressManually()]
+    ]);
+    const actionToPerform = actionsMap.get(action);
+    if (!actionToPerform) throw new Error(`No action found for '${action}'`);
+    await actionToPerform();
   }
 
-  async execute(page:Page,fieldName?: actionData, value?: actionData): Promise<void> {
-    if (!fieldName) throw new Error('Missing fieldName');
+  private async createCaseAction(page: Page, action: string, fieldName: actionData, data?: actionData) {
     const dataStoreApiInstance = await dataStoreApi();
-    await dataStoreApiInstance.execute(page, fieldName, value);
-    caseInfo = await dataStoreApiInstance.createCase(fieldName);
+    await dataStoreApiInstance.execute(page, action, fieldName, data);
+    caseInfo = await dataStoreApiInstance.createCase(fieldName as string);
+  }
+
+  private async selectAddress(caseData: actionData) {
+    const addressDetails = caseData as { postcode: string; addressIndex: number };
+    await performActions(
+      'Find Address based on postcode',
+      ['inputText', 'Enter a UK postcode', addressDetails.postcode],
+      ['clickButton', 'Find address'],
+      ['select', 'Select an address', addressDetails.addressIndex]
+    );
+    await performAction('clickButton', 'Continue');
+  }
+
+  private async selectLegislativeCountry(caseData: actionData) {
+    await performAction('clickRadioButton', caseData);
+    await performAction('clickButton', 'Continue');
+  }
+
+  private async selectClaimantType(caseData: actionData) {
+    await performAction('clickRadioButton', caseData);
+    await performAction('clickButton', 'Continue');
+  }
+
+  private async selectCaseOptions() {
```

Comment:
Can you rename it in a more meaningful way, such as before when it was selectJurisdictionCaseTypeEvent?